CMakeLists.txt

```diff
@@ -16,51 +16,29 @@ enable_testing()
 set (LUA_PATH "LUA_PATH=${PROJECT_SOURCE_DIR}/?.lua\\;${PROJECT_SOURCE_DIR}/?/init.lua\\;\\;")
 set (LUA_SOURCE_DIR "LUA_SOURCE_DIR=${PROJECT_SOURCE_DIR}")
 
-## VERSION ####################################################################
-###############################################################################
-
-if (NOT VERSION)
-  execute_process(
-    COMMAND git describe --tags --always
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-    OUTPUT_VARIABLE VERSION
-    ERROR_QUIET
-  )
-endif()
-
-if (VERSION)
-  configure_file (
-    "${PROJECT_SOURCE_DIR}/${PROJECT_NAME}/VERSION.lua.in"
-    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/VERSION.lua"
-  )
-elseif(NOT LUAROCKS) # allow luarocks to build without passing VERSION
-  message(FATAL_ERROR "VERSION is not provided")
-endif()
-
 ## Install ####################################################################
 ###############################################################################
 
-install(
-  FILES ${CMAKE_CURRENT_SOURCE_DIR}/cartridge-cli.lua
-  DESTINATION ${TARANTOOL_INSTALL_LUADIR}/
-)
-
-install(
-  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/templates
-            ${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}
-  USE_SOURCE_PERMISSIONS
-  DESTINATION ${TARANTOOL_INSTALL_LUADIR}/
-  PATTERN "*.in" EXCLUDE
+# Create tempdir
+execute_process(
+  COMMAND mktemp -d
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  OUTPUT_VARIABLE TMPDIR
 )
 
-install(
-  DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}
-  DESTINATION ${TARANTOOL_INSTALL_LUADIR}/
-  OPTIONAL
+# Generate cartridge binary in tmpdir
+execute_process(
```

Comment:
Seems we need something similar to https://cmake.org/cmake/help/v3.12/module/FindPython.html (available since cmake 3.12). You require 2.8 and older.

But the problem - you have different python versions on different distros and should try to find python3, python2 or throw an error.

Reply:
try to copy-paste it https://github.com/Kitware/CMake/blob/master/Modules/FindPython/Support.cmake

Reply:
I think, I can require cmake 3.

Reply:
> I think, I can require cmake 3.

No, because you supports CentOS 7

> It will be used on Gitlab CI, I can install any version I need there.

Yes, but my concern is about users who clone the repo and build/install cartridge-cli from source